diesel/src/pg/backend.rs

```diff
@@ -9,10 +9,57 @@ use prelude::Queryable;
 use query_builder::bind_collector::RawBytesBindCollector;
 use sql_types::{Oid, TypeMetadata};
 
+use std::slice;
+
 /// The PostgreSQL backend
 #[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
 pub struct Pg;
 
+/// The raw value representation of the postgres backend
+#[derive(Debug)]
+pub struct PgValue<'a> {
+    raw_bytes: &'a [u8],
+    oid: u32,
+}
+
+impl<'a> PgValue<'a> {
+    pub(crate) fn new(value_ptr: *mut u8, byte_count: usize) -> PgValue<'a> {
```

Comment:
Any reasons why we need a `*mut u8` and not a `*const u8` here?
```suggestion
    pub(crate) fn new(value_ptr: *const u8, byte_count: usize) -> PgValue<'a> {
```